libraries/abstractions/secure_sockets/utest/secure_sockets_utest.c

```diff
@@ -898,26 +898,82 @@ void test_SecureSockets_Connect_TLS_SNI_connect_error( void )
 /*!
  * @brief GetHostByName  successful case
  *
- * The purpose of this testase is to make sure sockets_gethostbyname is behaving
+ * The purpose of this test case is to make sure sockets_gethostbyname is behaving
  * properly under normal conditions and returning the expected data to the
  * caller
  */
-void test_SecureSockets_GetHostByName_suceessful( void )
+void test_SecureSockets_GetHostByName_successful( void )
 {
     int32_t ret;
     uint8_t addr = 0;
-    uint8_t ret_addr = 5;
+    uint32_t ret_addr = 5;
     int32_t hostnameMaxLen = securesocketsMAX_DNS_NAME_LENGTH;
     char hostname[ hostnameMaxLen ];
 
     strncpy( hostname, "this is a hostname", hostnameMaxLen );
 
-    WIFI_GetHostIP_ExpectAndReturn( hostname, &addr, 0 );
-    WIFI_GetHostIP_ReturnThruPtr_pucIPAddr( &ret_addr );
+    dns_gethostbyname_addrtype_ExpectAnyArgsAndReturn( ERR_OK );
+    dns_gethostbyname_addrtype_ReturnThruPtr_addr( &ret_addr );
     ret = SOCKETS_GetHostByName( hostname );
     TEST_ASSERT_EQUAL_INT( ret_addr, ret );
 }
 
+/*!
+ * @brief GetHostByName wait success case
+ *
+ * The purpose of this test case is to make sure sockets_gethostbyname
+ * handles the case where lwip dns must wait for completion.
+ */
+#define STUB_RETURNED_ADDRESS    5
+static err_t dns_gethostbyname_addrtype_success_CALLBACK( const char * hostname,
+                                                          ip_addr_t * addr,
+                                                          dns_found_callback found,
+                                                          void * callback_arg,
+                                                          u8_t dns_addrtype,
+                                                          int cmock_num_calls )
```

Comment:
I think there are some opportunities to make assertions on the parameters being passed to `dns_gethostbyname_addrtype`. For instance, checking `pcHostName` == `hostname`. Although, overly asserting to a point of following the source code too closely can be bad too.

Reply:
Agreed that we can further improve test coverage here, but the main purpose of the PR is to fix the currently broken test.  Additional improvements can be driven by tracking the unit test code coverage.